src/Main.cpp

```diff
@@ -111,24 +113,46 @@ struct EnemyMovementSystem
 
 	void on_tick() override
 	{
-		
-		
-		for (auto &&[player_entity, player_pos] : QueryPlayers::access_storage().each())
+		//enemy flocking system
+
+		auto player  = QueryPlayers::access_storage().front();
+		for (auto&& [entity, enemy, pos] : QueryEnemies::access_storage().each())
 		{
-			for (auto&& [entity, enemy, pos] : QueryEnemies::access_storage().each())
+			auto min_distance = 1000000;
+			auto nearest_enemy = Enemy();
```

Comment:
Unneeded values created here, we have raw pointers for exactly this kind of thing - set to `nullptr` at the start and then _point_ at the closest enemy.